src/lua/httpc.lua

```diff
@@ -313,6 +320,37 @@ local function encode_body(body, content_type, encoders)
     return raw_body
 end
 
+local default_decoder = function(raw_body, _content_type)
+    return json.decode(raw_body)
+end
+
+local function decode_body(response)
+    if response.body == nil then
+        error('Unable to decode body: body is empty')
+    end
+    local headers = response.headers or {}
+    local content_type = headers['content-type']
+    local mime_type = extract_mime_type(content_type)
```

Comment:
The same question about `Application/JSON` is relevant here, but I guess that it'll be fixed in the first commit by adjusting `extract_mime_type()`. I wrote this comment to don't miss it, just in case.